tests/defer/tests.py

```diff
@@ -49,8 +49,15 @@ def test_defer_only_chaining(self):
         qs = Primary.objects.all()
         self.assert_delayed(qs.only("name", "value").defer("name")[0], 2)
         self.assert_delayed(qs.defer("name").only("value", "name")[0], 2)
+        self.assert_delayed(qs.defer("name").only("name").only("value")[0], 2)
         self.assert_delayed(qs.defer("name").only("value")[0], 2)
         self.assert_delayed(qs.only("name").defer("value")[0], 2)
+        self.assert_delayed(qs.only("name").defer("name").defer("value")[0], 1)
+        self.assert_delayed(qs.only("name").defer("name", "value")[0], 1)
+
+        # Cancel each other out
+        self.assert_delayed(qs.only("name").defer("name")[0], 0)
+        self.assert_delayed(qs.defer("name").only("name")[0], 0)
```

Comment:
This works without the patch and should be moved to a separate commit. I'd also move it to a separate test method:
```python
def test_defer_only_clear(self):
    self.assert_delayed(qs.only('name').defer('name')[0], 0)
    self.assert_delayed(qs.defer('name').only('name')[0], 0)
```